napari/plugins/manager.py

```diff
@@ -0,0 +1,111 @@
+import importlib
+import os
+import pkgutil
+from logging import Logger
+
+import pluggy
+
+from . import _builtins, hookspecs
+
+logger = Logger(__name__)
+
+
+class NapariPluginManager(pluggy.PluginManager):
+    PLUGIN_ENTRYPOINT = "napari.plugin"
+    PLUGIN_PREFIX = "napari_"
+
+    def __init__(self, autodiscover=True):
+        """pluggy.PluginManager subclass with napari-specific functionality
+
+        In addition to the pluggy functionality, this subclass adds
+        autodiscovery using package naming convention.
+
+        Parameters
+        ----------
+        autodiscover : bool, optional
+            Whether to autodiscover plugins by naming convention and setuptools
+            entry_points, by default True
+        """
+        super().__init__("napari")
+
+        # define hook specifications and validators
+        self.add_hookspecs(hookspecs)
+
+        # register our own built plugins
+        self.register(_builtins, name='builtins')
+        # discover external plugins
+        if not os.environ.get("NAPARI_DISABLE_PLUGIN_AUTOLOAD"):
+            if autodiscover:
+                self.discover()
+
+    def discover(self):
```

Comment:
Perhaps there could be keyword arguments here for which modules can be discovered?

Reply:
added